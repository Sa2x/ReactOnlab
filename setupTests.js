import { render } from "@testing-library/react";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
configure({ adapter: new Adapter() });

global.renderWithRouter = (renderComponent, route) => {
  const history = createMemoryHistory();
  if (route) {
    history.push(route);
  }

  return {
    ...render(<Router history={history}>{renderComponent()}</Router>, history),
  };
};
