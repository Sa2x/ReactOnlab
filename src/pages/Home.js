import { motion } from "framer-motion";
import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faVideo } from "@fortawesome/free-solid-svg-icons";
const Home = () => {
  return (
    <div className="home">
      <div className="icons">
        {/* <FontAwesomeIcon icon={faVideo} />
        <FontAwesomeIcon icon={faVideo} />
        <FontAwesomeIcon icon={faVideo} />
        <FontAwesomeIcon icon={faVideo} />
        <FontAwesomeIcon icon={faVideo} />
        <FontAwesomeIcon icon={faVideo} /> */}
      </div>
      <h1>Watcherr</h1>
      <h3>Keep track of your favourite TV Shows</h3>
    </div>
  );
};

export default Home;
