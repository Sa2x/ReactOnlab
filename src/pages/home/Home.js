import React from "react";
const Home = () => {
  return (
    <div className="home">
      <div className="icons">
        {/* <FontAwesomeIcon icon={faVideo} />
        <FontAwesomeIcon icon={faVideo} />
        <FontAwesomeIcon icon={faVideo} />
        <FontAwesomeIcon icon={faVideo} />
        <FontAwesomeIcon icon={faVideo} />
        <FontAwesomeIcon icon={faVideo} /> */}
      </div>
      <h1>Watcherr</h1>
      <h3>Keep track of your favourite TV Shows</h3>
    </div>
  );
};

export default Home;
