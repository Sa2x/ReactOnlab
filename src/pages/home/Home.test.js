import React from "react";
import { shallow } from "enzyme";
import Home from "./Home";

import "../../../setupTests";

let title = "Watcherr";
let description = "Keep track of your favourite TV Shows";

let wrapped = shallow(<Home></Home>);

describe("HomePage", () => {
  it("Home Page should render correctly", () => {
    expect(wrapped).toMatchSnapshot();
  });

  it("contains the main texts", () => {
    expect(wrapped.find("h1").text()).toEqual(title);
    expect(wrapped.find("h3").text()).toEqual(description);
  });
});
