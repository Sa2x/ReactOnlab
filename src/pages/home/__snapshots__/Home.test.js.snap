// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Home Page should render correctly 1`] = `
<div
  className="home"
>
  <div
    className="icons"
  />
  <h1>
    Watcherr
  </h1>
  <h3>
    Keep track of your favourite TV Shows
  </h3>
</div>
`;
